app/screens/login/login.js

```diff
@@ -73,12 +74,13 @@ export default class Login extends PureComponent {
         setMfaPreflightDone(false);
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (this.props.loginRequest.status === RequestStatus.STARTED && nextProps.loginRequest.status === RequestStatus.SUCCESS) {
-            this.props.actions.handleSuccessfulLogin().then(this.goToChannel);
-        } else if (this.props.loginRequest.status !== nextProps.loginRequest.status && nextProps.loginRequest.status !== RequestStatus.STARTED) {
-            this.setState({isLoading: false});
+    static getDerivedStateFromProps(nextProps, prevState) {
+        if (prevState.loginRequest.status !== nextProps.loginRequest.status && nextProps.loginRequest.status !== RequestStatus.STARTED) {
+            return {isLoading: false};
+        } else if (nextProps.loginRequest.status === RequestStatus.STARTED && nextProps.loginRequest.status === RequestStatus.SUCCESS) {
+            return nextProps.actions.handleSuccessfulLogin().then(this.goToChannel);
```

Comment:
This is probably causing the test to fail. `this` is undefined in `getDerivedStateFromProps` since it's a static method, so this is probably failing silently which causes the test to just time out.

This logic should be moved to a `componentDidUpdate` anyway since it has side effects. `getDerivedStateFromProps` should only be used to update the state of the component based on the props, so we shouldn't be dispatching actions here. In `componentDidUpdate`, it'll also have access to `this` to switch the screen properly